crates/matrix-sdk/src/account.rs

```diff
@@ -916,6 +920,88 @@ impl Account {
         Ok(ignored_user_list)
     }
 
+    /// Set the media previews display policy for the account.
+    ///
+    /// This is an unstable event, so we always default to setting the unstable version of the event for now.
+    pub async fn set_media_previews_setting(&self, new_setting: MediaPreviews) -> Result<()> {
+        let mut media_preview_config_to_update = if let Some(unstable_media_preview_config) =
+            self.get_unstable_media_preview_config().await?
+        {
+            unstable_media_preview_config
+        } else {
+            UnstableMediaPreviewConfigEventContent::new(Default::default(), Default::default())
+        };
+        media_preview_config_to_update.media_previews = new_setting;
+        // Updating the account data
+        self.set_account_data(media_preview_config_to_update).await?;
+        // TODO: I think I should reset all the storage and perform a new local sync
+        // here but I don't know how
```

Comment:
Not really sure what to do here

Reply:
I was actually looking at this recently and was surprised to see the store not getting updated when we set some account data. @poljar is there a way to tell the store that a piece of account data has been changed?

Reply:
I just looked at the the `ignore_user` and `unignore_user` function which has the exact same comment. So I assume this is a known issue maybe? I imagine that since the sync might take some time this means that if we change the value and immediately check the account data again, it won't be updated right?
And if so how long may it take to be update, are we talking about seconds or milliseconds? 
Since this may affect how we design the UI.

Reply:
As long as a sync takes, you can subscribe to listen to account data events. There's actually a PR in flight to add exactly this to the bindings: https://github.com/matrix-org/matrix-rust-sdk/pull/4994.

I suppose you'll have to subscribe to this anyways, in case the value gets updated on another device.

Reply:
nice this would be very useful!